src/transformers/models/swin/modeling_swin.py

```diff
@@ -57,6 +60,71 @@
 ]
 
 
+@dataclass
+class SwinBaseModelOutput(ModelOutput):
+    """
+    Class for SwinEncoder's outputs.
+
+    Args:
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the model.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings + one for the output of each layer) of
+            shape `(batch_size, sequence_length, hidden_size)`.
+
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+        hidden_states_spatial_dimensions (`tuple(tuple(int, int))`, *optional*):
```

Comment:
Not sure whether it makes sense to do this. If the model just returns hidden states of shape (B, C, H, W), then we get both hidden states and spatial dimensions for free.

Reply:
+1 on this, very nice idea. 

From a design point of view, if we say `output_hidden_states` the user will expect to receive the `hidden_states` untouched. But I see why this is more convenient. 

Pinging @sgugger for a feedback

Reply:
I have already suggested this on the Maskformer PR but it was ignored so I thought it wasn't possible. If it indeed is, it would kill two birds with one stone.

Reply:
Apologies if it was ignored in MaskFormer. it is indeed possible but requires some refactoring. let me try to change this